feat(vscode): add attach mode for connecting to existing server

Destreyf opened the pull request:
## Summary
- Add configuration settings for attach mode (`opencode.attach.enabled`, `.host`, `.port`)
- When enabled, run `opencode attach` instead of spawning a new instance
- Support both `http://` and `https://` protocols in host configuration
- Enables session sharing between terminal/web and VSCode in cloud workspaces (e.g., Coder)

## Changes
- `sdks/vscode/package.json`: Add `contributes.configuration` section
- `sdks/vscode/src/extension.ts`: Read config, conditional attach command, protocol-aware URL building
- `sdks/vscode/README.md`: Document new settings including HTTPS support

## Usage
Enable attach mode in VS Code settings:
```json
{
  "opencode.attach.enabled": true,
  "opencode.attach.host": "localhost",
  "opencode.attach.port": 4096
}
```

For HTTPS connections:
```json
{
  "opencode.attach.host": "https://myserver.com",
  "opencode.attach.port": 443
}
```

Destreyf committed:
feat(vscode): add attach mode for connecting to existing server

rekram1-node commented:
/review

Destreyf committed:
Apply suggestions from code review

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Destreyf committed:
Merge branch 'dev' into feat/vscode-attach-mode

Destreyf commented:
@rekram1-node I should have definitely had a little more attention before I submitted this.  It caught a legitimate potential bug. 

I've updated it with the suggested fix, it's slightly different syntax than the way the original did it, but it should accomplish the same outcome. 

If you want me to adjust the formatting let me know.

Destreyf committed:
Merge branch 'dev' into feat/vscode-attach-mode

adamdotdevin committed:
feat(desktop): new layout

adamdotdevin committed:
fix(desktop): layout

Commit:
release: v1.0.181

rekram1-node committed:
fix: regression where config would error despite valid agents

actions-user committed:
chore: generate

Commit:
release: v1.0.182

rekram1-node commented:
this is mostly just the builtin /review command nothing super crazy glad it works!

rekram1-node commented:
/review

rekram1-node committed:
core: add verification that at least 1 primary agent is enabled, add regression tests  (#5881)

christso committed:
Make CLI build script Windows-friendly (#5835)

actions-user committed:
Update Nix flake.lock and hashes

adamdotdevin committed:
fix(desktop): file loading errors

adamdotdevin committed:
fix(desktop): non-latin file paths failed

adamdotdevin committed:
fix(desktop): better error reporting

Commit:
release: v1.0.183

adamdotdevin committed:
fix(desktop): incorrect state dir on macos

actions-user committed:
chore: generate

adamdotdevin committed:
fix(desktop): allow text selection

adamdotdevin committed:
fix(desktop): todo tool title

adamdotdevin committed:
fix(desktop): better keybind tooltips

adamdotdevin committed:
fix(desktop): layout regression

Commit:
release: v1.0.184

actions-user committed:
ignore: update download stats 2025-12-21

elithrar committed:
github: support schedule events (#5810)

actions-user committed:
chore: generate

seuros committed:
fix: use official MCP SDK for better tool schema handling (#5463)

ahosker committed:
fix: prevent stats workflow from running on forks (#5897)

nalin-singh committed:
feat: add syntax highlighting for .ets files (#5889)

HelloGGX committed:
fix(deps): add missing @opencode-ai/plugin to dependencies (#5797)

actions-user committed:
chore: generate

actions-user committed:
Update Nix flake.lock and hashes

thdxr committed:
tui: fix SDK context usage and server port fallback

- Update SDK context to return client instead of event for proper usage
- Add server port fallback to 4096 when port 0 is specified but unavailable
- Fix SDK event listener usage in TUI app

rekram1-node committed:
tweak: update kimi-k2 and kimi-k2-thinking to use recommended temperature values

thdxr committed:
core: fix LSP server binary installation and shell command execution

- Ensure proper file permissions are set for installed LSP binaries on non-Windows platforms
- Add error handling for shell command execution in prompt system to prevent crashes

Commit:
release: v1.0.185

Destreyf committed:
fix: cleanup merge isuse and update to match style guide.

Destreyf committed:
Merge branch 'dev' into feat/vscode-attach-mode

Destreyf commented:
/review

(does this work for non contributers, i doubt it)

Destreyf committed:
feat(vscode): support https protocol in attach mode host

Destreyf commented on sdks/vscode/src/extension.ts:
```diff
@@ -34,17 +34,35 @@ export function activate(context: vscode.ExtensionContext) {
+      }
+      terminal.show()
       terminal.show()
     }
```
Not sure why it still shows this here, but i have updated the code since then.

rekram1-node commented:
it only works for admins!

rekram1-node commented:
/review

Destreyf commented:
> it only works for admins!

I figured as much, I am actually using your guy's exact review workflow in a project of mine, and will be adding it to the rest of my repo's.  I saw "Owner" as a filter but wasn't certain on who counted as that fronm my brief skim.

The primary motivator for this update to the vscode extension is I am running Coder workspaces and I run a copy of opencode as a background task, so being able to attach to it is super helpful from within vscode.

My normal workflow currently is to "attach" to the server outside of the vscode extension, but this makes it much easier to have a one click launch, the actual endpoint for us is always going to be localhost:4096, but having it configurable could be beneficial if you're running something like cloudflare agents to host a copy of opencode, or really anytime you could want to attach to a running instance.

LMK if there's any concerns.

rekram1-node commented on sdks/vscode/package.json:
```diff
@@ -78,7 +78,27 @@
+    "configuration": {
+      "title": "OpenCode",
+      "properties": {
+        "opencode.attach.enabled": {
+          "type": "boolean",
+          "default": false,
+          "description": "Attach to an existing OpenCode server instead of spawning a new instance"
+        },
```
if u pass attach.host or attach.port, what purpose does this bool serve? Wouldn't setting attach.host / attach.post infer it?

Destreyf replied:
I went back and forth on this, honestly the only purpose it would serve is to act as a "quick disable" so you didn't have to remove the values from host & port.  But I also don't know that we need to keep it.

rekram1-node replied:
I think we should drop it if we can help it?

Destreyf replied:
We can definitely drop it, I have a "working" one in my local branch, but in my testing I ran into a couple hitches.

I'll elaborate in a followup comment.

Destreyf commented:
The way the extension currently works is when the extension opens a terminal with a random port, so it can communicate for the append.  

I don't see an easy endpoint for doing this same thing, so the solution I am testing is using the `terminal.sendText` with the `shouldExecute` set to false.

This really only impacts if a user is opening multiple opencode windows inside of the editor.

If we need to have a larger discussion around this I am happy to put this on hold.

rekram1-node commented:
wait so what does that have to do w/ the extra setting? I don't know if im following
